appium/options/android/common/locking/locking_options.py

```diff
@@ -0,0 +1,165 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Optional, Union
+
+from appium.options.base_options_descriptor import OptionsDescriptor
+from appium.options.common.supports_capabilities import SupportsCapabilities
+from appium.options.transformers import transform_duration_get, transform_duration_set
+
+
+class SkipUnlockOption(SupportsCapabilities):
+    SKIP_UNLOCK = "skipUnlock"
+    skip_unlock = OptionsDescriptor[Optional[bool], bool](SKIP_UNLOCK)
+    """
+    Whether to skip the check for lock screen presence (true). By default,
+    the driver tries to detect if the device's screen is locked
+    before starting the test and to unlock that (which sometimes might be unstable).
+    Note, that this operation takes some time, so it is highly recommended to set
+    this capability to true and disable screen locking on devices under test.
+
+    Usage
+    -----
+    - Get
+        - `self.skip_unlock`
+    - Set
+        - `self.skip_unlock` = `value`
+    
+    Parameters
+    ----------
+    `value`: `bool`
+
+    Returns
+    -------
+    - Get
+        - `Optional[bool]`
+    - Set
+        - `None`
+    """
+
+
+class UnlockKeyOption(SupportsCapabilities):
+    UNLOCK_KEY = "unlockKey"
+    unlock_key = OptionsDescriptor[Optional[str], bool](UNLOCK_KEY)
```

Comment:
is the incoming value of type boolean for this cap?

Reply:
My mistake.. I have fixed this. It should be a `str`